File: 1 Revalida en python/5 Funciones/7.py
#funcion del algoritmo Honer
def honer(a,k):
	y=1
	while (k>0):
		if((k%2)==1):
			y=y*a
		a=a*a
		k=(k/2)
	return y
a=int(input("ingrese el numero de la base:"))
k=int(input("ingrese un numero de la exponete:"))
print honer(a,k)

File: 2 Revalida en C++/5 Funciones/7.cpp
///funcion del algoritmo Honer
#include <bits/stdc++.h>
using namespace std;
int honer(int a,int k)
{
    int y=1;
    while (k>0)
		{
		    if((k%2)==1)
			{
			    y=y*a;
                a=a*a;
                k=(k/2);
			}
		}
	return y;
}
int main()
{
    int a,k;
    cout<<"Ingrese a:";
    cin>>a;
    cout<<"Ingrese k:";
    cin>>k;
    cout<<"El resultado es ="<<honer(a,k);
    return 0;
}
